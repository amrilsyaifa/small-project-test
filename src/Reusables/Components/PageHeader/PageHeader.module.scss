@import 'src/Reusables/Styles/Variables';

.container {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  padding-left: 30px;
  padding-right: 30px;
}
.wrapper-right-text {
  font-size: 14px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  color: $grey;
  margin-top: 10px;
  .wrapper-text {
    width: 150px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    .text {
      cursor: pointer;
    }
  }
}

.header-wrapper {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 70px;
  img {
    width: 70px;
    height: 70px;
    cursor: pointer;
  }
  .search {
    margin-top: 15px;
    margin-left: 30px;
  }
}
