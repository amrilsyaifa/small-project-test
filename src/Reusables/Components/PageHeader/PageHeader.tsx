import { Typography } from 'antd';
import React from 'react';
import Styles from './PageHeader.module.scss';
import { useLocation } from 'wouter';

const { Text } = Typography;

const PageHeader: React.FC = () => {
  const [, go] = useLocation();

  return (
    <div className={Styles['container']}>
      <div className={Styles['header-wrapper']}>
        <img src="/logo512.png" alt="logo" onClick={() => go('/')} />
      </div>
      <div className={Styles['wrapper-right-text']}>
        <div className={Styles['wrapper-text']}>
          <Text className={Styles['text']}>Login</Text>
          <Text className={Styles['text']}>Logout</Text>
        </div>
      </div>
    </div>
  );
};

export default PageHeader;
