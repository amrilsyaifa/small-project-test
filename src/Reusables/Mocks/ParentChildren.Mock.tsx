import React from 'react';

/**
 * Separate children from Component so it's not circular referencing when stringified
 */
interface IParentChildrenSeparator {
  mockLabel: string;
  children: React.ReactNode;
  MockElement?: React.FC;
}
export const ParentChildrenSeparator: React.FC<IParentChildrenSeparator> = ({
  mockLabel,
  children,
  MockElement,
  ...props
}) => {
  const RenderElement = MockElement ? MockElement : MockDivElement;

  return (
    <RenderElement data-testid={mockLabel} data-props={JSON.stringify(props)}>
      {children}
    </RenderElement>
  );
};

export const MockDivElement = ({ ...props }) => <div {...props} />;

export default ParentChildrenSeparator;
