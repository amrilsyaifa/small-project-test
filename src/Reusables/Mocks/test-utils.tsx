import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { persistedReducer } from 'src/Store';
// Import your own reducer

interface IWrapper {
  children: React.ReactNode;
}

const render = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  { store = configureStore({ reducer: persistedReducer }), ...renderOptions } = {}
) => {
  const Wrapper: React.FC<IWrapper> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
