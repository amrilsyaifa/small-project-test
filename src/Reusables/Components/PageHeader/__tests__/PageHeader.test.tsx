import { render, screen, cleanup } from 'src/Reusables/Mocks/test-utils';
import React from 'react';
import PageHeader from '../PageHeader';
import { MatchMediaObject } from 'src/Reusables/Mocks/MatchMediaObjectTest';

afterEach(cleanup);
beforeEach(() => {
  MatchMediaObject();
});

describe('PageHeader', () => {
  test('PageHeader Render Snapshoot', () => {
    const { container } = render(<PageHeader />);
    expect(container).toMatchSnapshot();
  });

  test('PageHeader Have Title', () => {
    render(<PageHeader />);
    const titleHeader = screen.getByText('Login');
    expect(titleHeader).toBeInTheDocument();
  });
});
