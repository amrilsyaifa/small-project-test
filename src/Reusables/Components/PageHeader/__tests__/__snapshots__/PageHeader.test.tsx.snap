// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader PageHeader Render Snapshoot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header-wrapper"
    >
      <img
        alt="logo"
        src="/logo512.png"
      />
    </div>
    <div
      class="wrapper-right-text"
    >
      <div
        class="wrapper-text"
      >
        <span
          class="ant-typography text"
        >
          Login
        </span>
        <span
          class="ant-typography text"
        >
          Logout
        </span>
      </div>
    </div>
  </div>
</div>
`;
